src/misc/matching.c

```diff
@@ -1007,6 +1007,450 @@ static igraph_error_t igraph_i_maximum_bipartite_matching_weighted(
+                                            debug("found blossom root\n");
+                                        } else {
+                                            VECTOR(b_mark)[current] = true;
+                                        }
+                                    } else {
+                                        VECTOR(b_mark)[current] = false;
+                                    }
+                                }
+                                igraph_vector_int_t *list = igraph_vector_int_list_get_ptr(&b_lists, blossom_root);
+
+                                // mark all vertices in B as contracted to blossom root
+                                // do by traversing again, but stopping when blossom marks stop
+                                current = v;
+                                while (current != -1 && VECTOR(b_mark)[current]) {
+                                    VECTOR(contraction)[current] = blossom_root;
+                                    VECTOR(b_mark)[current] = false;
+                                    igraph_vector_int_push_back(list, current);
+                                    current = VECTOR(f_parent)[current];
+                                }
+                                current = w;
+                                igraph_vector_int_resize(&temp_list, 0);
+                                while (current != -1 && VECTOR(b_mark)[current]) {
+                                    VECTOR(contraction)[current] = blossom_root;
+                                    VECTOR(b_mark)[current] = false;
+                                    igraph_vector_int_push_back(&temp_list, current);
+                                    current = VECTOR(f_parent)[current];
+                                }
+                                igraph_vector_int_reverse(&temp_list);
+                                igraph_vector_int_append(list, &temp_list);
+                            }
+                        }
+                    }
+                    // mark edge e
+                    MATRIX(e_mark, v, w) = true;
+                }
+            }
+            // mark vertex v
+            VECTOR(v_mark)[v] = true;
+        }
+    }
+    //destroy memory things at end
+    igraph_vector_int_destroy(&f_parent);
+    igraph_vector_bool_destroy(&f_is_root);
+    igraph_vector_int_destroy(&root);
+    igraph_vector_int_destroy(&distance);
+    igraph_vector_int_destroy(&contraction);
+    igraph_vector_int_destroy(&neighbors);
+    igraph_dqueue_int_destroy(&queue);
+    igraph_vector_bool_destroy(&v_mark);
+    igraph_vector_bool_destroy(&b_mark);
+    igraph_matrix_bool_destroy(&e_mark);
+    igraph_vector_int_list_destroy(&b_lists);
+    igraph_vector_int_destroy(&temp_list);
+    IGRAPH_FINALLY_CLEAN(12);
+
+    return IGRAPH_SUCCESS;
+}
+
+void igraph_i_maximum_matching_get_tree_path(igraph_integer_t *current, igraph_vector_bool_t *f_is_root, igraph_vector_int_t *contraction,
```

Comment:
This function is `void` instead of returning `igraph_error_t` and the `push_back()` calls within are not checked for errors. Are we certain that these `push_back()` calls cannot fail, i.e. are they acting on a vector with guaranteed sufficient reserved storage? If yes, this should be explained in a comment. If not, error checks are necessary.